reconcile/utils/saasherder.py

```diff
@@ -794,3 +794,25 @@ def update_config(self, job_spec):
         key = f"{saas_file_name}/{rt_name}/{cluster_name}/" + \
             f"{namespace_name}/{env_name}"
         self.state.add(key, value=target_config, force=True)
+
+    def validate_promotions(self):
+        """
+        If there were promotion sections in the participating saas files
+        validate that the conditions are met. """
+        for item in self.promotions:
+            # validate that the commit sha being promoted
+            # was succesfully published to the subscribed channel(s)
+            commit_sha = item['commit_sha']
+            subscribe = item.get('subscribe')
+            if subscribe:
+                for channel in subscribe:
+                    state_key = f"{channel}/{commit_sha}"
```

Comment:
I believe we need a common prefix path, otherwise the state will be a mess of keys.

Reply:
like this?
https://github.com/app-sre/qontract-reconcile/blob/58ba6944dd61c810e90e1775e5078c572691a605/reconcile/utils/state.py#L28

Reply:
and each integration "chooses" how it handles it's own "space"

Reply:
I was thinking in something like `state_key = f"promotions/{channel}/{commit_sha}"`, but whater aggregates the promotion state keys in the first level would work, I guess

Reply:
I'm also trying to avoid overriding keys that belong to other features

Reply:
`promotions/{channel}/{commit_sha}` makes sense. also didn't think about multiple features in the same integration requiring the use of state. all good points!